okio/src/main/java/okio/GzipSource.java

```diff
@@ -68,6 +68,10 @@ public GzipSource(Source source) {
     if (byteCount < 0) throw new IllegalArgumentException("byteCount < 0: " + byteCount);
     if (byteCount == 0) return 0;
 
+    if (source.exhausted()) {
```

Comment:
While it may be malformed in the context of HTTP, I'm not sure passing something empty through an `GzipSource` that happens to be empty is malformed (e.g., a `File`) and we should rather adhere to the `Source` spec.